Feature/lib 34

Lelik7777 opened the pull request:
# Pagination Component Implementation

## Changes
This PR introduces a new pagination component with the following features:

1. Custom pagination hook (`usePagination`) for handling pagination logic
2. Reusable `Pagination` component with customizable page size and navigation
3. `PaginationButton` component for individual page buttons
4. `SelectContainer` for page size selection
5. Storybook stories showcasing default and select-enabled pagination

Additionally, this PR includes some project maintenance:
- Removed the Vite config timestamp file

## Implementation Details
- The `usePagination` hook calculates the pagination range and provides navigation functions
- The `Pagination` component uses the hook to render a fully functional pagination UI
- `PaginationButton` is used for individual page numbers and navigation buttons
- `SelectContainer` allows users to change the number of items per page
- Storybook stories demonstrate the component's usage and variations

## Testing
- Storybook stories have been added to showcase and test the component's functionality
- Manual testing has been performed to ensure correct behavior in various scenarios

## Notes for Reviewers
- Please pay attention to the pagination logic in `usePagination` hook
- Check if the component is accessible and works well with keyboard navigation
- Verify that the Storybook stories cover all main use cases


## Screenshots
![Screenshot_3](https://github.com/user-attachments/assets/58516a3b-0a32-46fb-9f83-2b5396552e9a)

Lelik7777 committed:
refactor: delete vite.config.timestamp

Lelik7777 committed:
feature(pagination): add pagination folder with templates

Lelik7777 committed:
feat(pagination): implement custom pagination component with hooks

- Add usePagination hook for pagination logic
- Create Pagination component with customizable page size and navigation
- Implement PaginationButton for individual page buttons
- Add SelectContainer for page size selection
- Include Storybook stories for default and select-enabled pagination

Lelik7777 committed:
fix(pagination): ensure all code paths return a value in usePagination hook

- Removed unnecessary curly braces in if statements for clearer returns.
- Added a default return statement to ensure the pagination range is always returned.

NoName-ForTeam commented on src/components/ui/pagination/pagination.tsx:
```diff
@@ -0,0 +1,163 @@
+  siblingCount = 1,
+  totalCount,
+}: PaginationProps) => {
+  const classNames = { container: styles.container, root: styles.root } as const
```
думаю, нужно добавить className и ...rest для главного контейнера

Lelik7777 replied:
добавил и исправил

Lelik7777 committed:
refactor(Pagination): add forwardRef, className, and rest props

- Wrap PaginationButton and SelectContainer components with forwardRef
- Add className prop to PaginationButton and SelectContainer
- Spread rest props in PaginationButton and SelectContainer
- Update types to include ComponentPropsWithoutRef

Lelik7777 committed:
Merge branch 'dev' into feature/lib-34
